### PR TITLE
Successfully implemented swap feature

### PR DESCRIPTION
**Pull request by RamanSB**

### CODE REVIEW

File: curve-fi/src/components/SwapPane.jsx [hunk 1]
```diff
@@ -28,69 +44,191 @@ function SwapPane() {
...
+                    x.getTicker() === parentState.fromCurrency
+                  ) {
+                    return x.getImgPath();
+                  } else if (
+                    label === "to" &&
+                    x.getTicker() === parentState.toCurrency
+                  ) {
+                    return x.getImgPath();
+                  }
+                  return null;
```
Comment (RamanSB) [inline curve-fi/src/components/SwapPane.jsx:173]: shouldnt be doing this..

File: curve-fi/src/components/SwapPane.jsx [hunk 2]
```diff
@@ -18,6 +18,22 @@ function SwapPane() {
     toAmount: 0,
     toCurrency: "USDC",
   });
+  console.log(initialState);
```
Comment (darrelladjei) [inline curve-fi/src/components/SwapPane.jsx:21]: is this log statement needed

File: curve-fi/src/components/SwapPane.jsx [hunk 3]
```diff
@@ -28,69 +44,191 @@ function SwapPane() {
...
+              }}
+            />
             <img
               src={swapIcon}
               alt="swap"
               width="36px"
+              height="36px"
               style={{ margin: "0 12px 0 6px" }}
+              onClick={() => {
+                console.log(`Hi!`);
```
Comment (darrelladjei) [inline curve-fi/src/components/SwapPane.jsx:66]: lol man said "hi" in submitted code

File: curve-fi/src/components/SwapPane.jsx (hunk 2 above)
Comment (RamanSB) [reply curve-fi/src/components/SwapPane.jsx:21]: Nope, even you commenting on this makes me feel like we are going to make great progress.

File: curve-fi/src/components/SwapPane.jsx [hunk 4]
```diff
@@ -28,69 +44,191 @@ function SwapPane() {
...
+              alignItems: "center",
+              width: "140px",
+              height: "30px",
+            }}
+          >
+            <img
+              width="24px"
+              height="24px"
+              src={cryptoCurrencies
+                .find((x) => {
```
Comment (darrelladjei) [inline curve-fi/src/components/SwapPane.jsx:161]: nit: .find((currency) => ...

File: curve-fi/src/components/SwapPane.jsx [hunk 5]
```diff
@@ -28,69 +44,191 @@ function SwapPane() {
...
+                  return null;
+                })
+                .getImgPath()}
+              alt={cryptoCurrencies[selectedOption].getTicker()}
+              style={{ margin: "0 16px 0 0" }}
+            />
+            {cryptoCurrencies
+              .find((x) => {
+                if (
+                  label === "from" &&
```
Comment (darrelladjei) [inline curve-fi/src/components/SwapPane.jsx:182]: nit: maybe have these labels as consts

File: curve-fi/src/components/SwapPane.jsx [hunk 6]
```diff
@@ -28,69 +44,191 @@ function SwapPane() {
...
+              .getTicker()}
+            <i class="fa-solid fa-angle-down" style={{ flexGrow: 1 }}></i>
+          </div>
+        </button>
+        <ul
+          className={`options ${isOptionsOpen ? "show" : ""}`}
+          tabIndex={-1}
+          role="listbox"
+        >
+          {cryptoCurrencies.map((option, index) => (
```
Comment (darrelladjei) [inline curve-fi/src/components/SwapPane.jsx:202]: .map((currency, index)

**Review by darrelladjei — CHANGES_REQUESTED** [review]

**Pull request merged**